Frameworks/UIKit/UIViewController.mm

```diff
@@ -1250,18 +1256,29 @@ - (void)dismissViewControllerAnimated:(BOOL)animated completion:(void (^)(void))
         curController->priv->_parentViewController->priv->_modalViewController = nil;
     }
```

Comment:
I have some changes in #1511 making the _presentedViewController a StrongId and some related changes that better handle the cleanup of the modal you've pointed to. I can put those changes into this commit as well.